src/Keboola/DbWriter/Snowflake/Connection.php

```diff
@@ -149,6 +166,35 @@ public function getTablePrimaryKey($schemaName, $tableName)
         return $pkCols;
     }
 
+    public function getTableUniqueKeys(string $schemaName, string $tableName): array
+    {
+        $cols = $this->fetchAll(sprintf(
+            'DESC TABLE %s.%s',
+            $this->quoteIdentifier($schemaName),
+            $this->quoteIdentifier($tableName)
+        ));
+        $uniqueCols = [];
+        foreach ($cols as $col) {
+            if ($col['unique key'] !== 'Y') {
+                continue;
+            }
+            $uniqueCols[] = $col['name'];
+        }
+
+        return $uniqueCols;
+    }
+
+    public function checkTableConstraints(string $schemaName, string $tableName, string $type = 'FOREIGN KEY'): bool
+    {
+        $rows = $this->fetchAll(sprintf(
+            "SELECT * FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS WHERE TABLE_SCHEMA = '%s' AND TABLE_NAME = '%s' AND CONSTRAINT_TYPE = '%s'",
+            $schemaName,
+            $tableName,
+            $type
+        ));
+        return count($rows) > 0;
```

Comment:
Bohužel je to jen na počet cizích klíčů v tabulce - snowflake jaksi zapomněl ukazatel konkrétního sloupce do `INFORMATION_SCHAMA`

Reply:
viz: https://github.com/keboola/db-writer-snowflake/pull/63#discussion_r331886199

Reply:
už tam něco podobného řešili:
https://support.snowflake.net/s/question/0D50Z00008Y3VUbSAN/table-details-including-the-constraints

Reply:
Přidal jsem ping, jestli už to neplánují opravit.